src-java/testing/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/flows/FlowHistorySpec.groovy

```diff
@@ -135,140 +138,146 @@ class FlowHistorySpec extends HealthCheckSpecification {
 
         and: "Flow history statuses returns all flow statuses for the whole life cycle"
         //create, update, delete
-        northboundV2.getFlowHistoryStatuses(flow.flowId).historyStatuses*.statusBecome == ["UP", "UP", "DELETED"]
+        updatedFlow.retrieveFlowHistoryStatus()*.statusBecome == [UP, UP, DELETED]
         //check custom timeLine and the 'count' option
-        northboundV2.getFlowHistoryStatuses(flow.flowId, specStartTime, timestampAfterUpdate)
-                .historyStatuses*.statusBecome == ["UP", "UP"]
-        northboundV2.getFlowHistoryStatuses(flow.flowId, 1).historyStatuses*.statusBecome == ["DELETED"]
+        updatedFlow.retrieveFlowHistoryStatus(specStartTime, timestampAfterUpdate)*.statusBecome == [UP, UP]
+        updatedFlow.retrieveFlowHistoryStatus(1)*.statusBecome == [DELETED]
     }
 
     def "History records are created for the create/update actions using default timeline"() {
         when: "Create a flow"
         def (Switch srcSwitch, Switch dstSwitch) = topology.activeSwitches
-        def flow = flowHelperV2.randomFlow(srcSwitch, dstSwitch)
-        flowHelperV2.addFlow(flow)
+        def flow = flowFactory.getRandom(srcSwitch, dstSwitch)
 
         then: "History record is created"
-        def flowHistory = northbound.getFlowHistory(flow.flowId)
-        assert flowHistory.size() == 1
-        checkHistoryCreateAction(flowHistory[0], flow.flowId)
+        flow.waitForHistoryEvent(FlowActionType.CREATE)
+        def flowHistory = flow.retrieveFlowHistory()
+        assert flowHistory.entries.size() == 1
+        checkHistoryCommonStuff(flowHistory, flow.flowId, FlowActionType.CREATE)
 
         when: "Update the created flow"
-        flowHelperV2.updateFlow(flow.flowId, flow.tap { it.description = it.description + "updated" })
+        flow.update(flow.tap {it.description = it.description + "updated" })
 
         then: "History record is created after updating the flow"
-        def flowHistory1 = northbound.getFlowHistory(flow.flowId)
-        assert flowHistory1.size() == 2
-        checkHistoryUpdateAction(flowHistory1[1], flow.flowId)
+        flow.waitForHistoryEvent(FlowActionType.UPDATE)
+        def flowHistory1 = flow.retrieveFlowHistory()
+        assert flowHistory1.entries.size() == 2
+        checkHistoryCommonStuff(flowHistory1, flow.flowId, FlowActionType.UPDATE)
 
         when: "Delete the updated flow"
-        flowHelperV2.deleteFlow(flow.flowId)
+        flow.delete()
 
         then: "History is still available for the deleted flow"
-        def flowHistory3 = northbound.getFlowHistory(flow.flowId)
-        assert flowHistory3.size() == 3
-        checkHistoryDeleteAction(flowHistory3, flow.flowId)
+        flow.waitForHistoryEvent(FlowActionType.DELETE)
+        def flowHistory3 = flow.retrieveFlowHistory()
+        assert flowHistory3.entries.size() == 3
+        checkHistoryCommonStuff(flowHistory3, flow.flowId, FlowActionType.CREATE)
```

Comment:
Just as idea, maybe it is easier to check the list of action types and verify for each action inside the method, like
```
checkHistoryCommonStuff(flowHistory3, flow.flowId, [FlowActionType.CREATE, FlowActionType.UPDATE, FlowActionType.DELETE])
```
But this is up to you, please ignore if it is not applicable or comfortable